partner_contact_in_several_companies/tests/test_partner_contact_in_several_companies.py

```diff
@@ -1,262 +1,217 @@
+            u'Bob Egnops'
         )
 
         # Reset contact to standalone
-        self.partner.write(cr, uid, [new_contact_id], {'contact_id': False})
+        new_contact.write({'contact_id': False})
         self.assertEqual(
-            self.partner.browse(cr, uid, new_contact_id).contact_type,
-            'standalone',
+            new_contact.contact_type,
+            u'standalone',
         )
 
         # Reset contact to attached, and ensure only it is unlinked (i.e.
         # context is ignored).
-        self.partner.write(cr, uid, [new_contact_id],
-                           {'contact_id': self.bob_contact_id})
+        new_contact.write({'contact_id': self.bob_contact.id})
         ctx = {'search_show_all_positions': {'is_set': True,
                                              'set_value': True
                                              }}
-        self.partner.unlink(cr, uid, [new_contact_id], context=ctx)
-        partner_ids = self.partner.search(
-            cr, uid, [('id', 'in', [new_contact_id, self.bob_contact_id])])
-        self.assertIn(self.bob_contact_id, partner_ids)
-        self.assertNotIn(new_contact_id, partner_ids)
+        new_contact.with_context(ctx).unlink()
+        partner_ids = self.partner.with_context(ctx).search(
+            [('id', 'in', [new_contact.id, self.bob_contact.id])])
+        self.assertIn(self.bob_contact, partner_ids)
+        self.assertNotIn(new_contact, partner_ids)
 
     def test_05_contact_fields_sync(self):
         """Check that contact's fields are correctly synced between
         parent contact or related contacts
         """
-        cr, uid = self.cr, self.uid
 
         # Test DOWNSTREAM sync
-        self.partner.write(
-            cr, uid, [self.bob_contact_id], {'name': 'Rob Egnops'}
+        self.bob_contact.write(
+            {'name': 'Rob Egnops'}
         )
         self.assertEqual(
-            self.partner.browse(cr, uid, self.bob_job1_id).name,
+            self.bob_job1.name,
             'Rob Egnops',
         )
 
         # Test UPSTREAM sync
-        self.partner.write(cr, uid, [self.bob_job1_id], {'name': 'Bob Egnops'})
+        self.bob_job1.write({'name': 'Bob Egnops'})
         self.assertEqual(
-            self.partner.browse(cr, uid, self.bob_contact_id).name,
+            self.bob_contact.name,
             'Bob Egnops',
         )
 
     def test_06_ir_action(self):
         """Check ir_action context is auto updated.
         """
-        cr, uid = self.cr, self.uid
 
         new_context_val = "'search_show_all_positions': " \
             "{'is_set': True, 'set_value': False},"
+        details = self.env.ref('base.action_partner_form')
 
-        details = self.action.read(
-            cr, uid, [self.base_partner_action_id]
-        )
         self.assertIn(
             new_context_val,
-            details[0]['context'],
+            details.context,
             msg='Default actions not updated with new context'
```

Comment:
This test fails because the context is modified in the `read` method only.
I think this can be considered as a bug (only changing values in the `read` method), but I don't find a way to change the value for a recordset.